### PR TITLE
feat: support ISO8601 formatted strings in timestamp column

### LINKED ISSUE #592: [ENHANCEMENT] support ISO formatted strings in timestamp column
**Is your feature request related to a problem? Please describe.**
Some users have shown they use ISO-formatted strings as their primary way to represent timestamps.

**Describe the solution you'd like**
Support ISO-formatted strings in timestamp column.

**Describe alternatives you've considered**
We could not support and require the user to convert to pandas datatime.

### PR DESCRIPTION
**Pull request by pbadhe**
Resolves #592 
1. Object dtype timestamp support and parameterized tests added, old invalid test modified.
2. Documentation updated.
3. Windows-friendly quotes change in phoenix dev install command. (Works in UNIX too)

### CODE REVIEW

**Comment by pbadhe** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by pbadhe** [comment]
recheck

File: tests/datasets/test_dataset.py [hunk 1]
```diff
@@ -1092,9 +1287,9 @@ def test_normalize_timestamps_raises_value_error_for_invalid_input() -> None:
             dataframe=DataFrame(
                 {
                     "timestamp": [
-                        "2023-03-24 10:00:00 UTC",
-                        "2023-03-24 11:30:00 UTC",
-                        "2023-03-24 14:15:00 UTC",
+                        "24-03-2023 invalidCharacter 14:15:00",
+                        "24-03-2023 invalidCharacter 10:30:00",
+                        "24-03-2023 invalidCharacter 18:45:00",
```
Comment (mikeldking) [inline tests/datasets/test_dataset.py:1292]: Mistaken checkin? `invalidCharacter`

File: tests/datasets/test_dataset.py [hunk 2]
```diff
@@ -1092,9 +1287,9 @@ def test_normalize_timestamps_raises_value_error_for_invalid_input() -> None:
             dataframe=DataFrame(
                 {
                     "timestamp": [
-                        "2023-03-24 10:00:00 UTC",
-                        "2023-03-24 11:30:00 UTC",
-                        "2023-03-24 14:15:00 UTC",
+                        "24-03-2023 invalidCharacter 14:15:00",
```
Comment (mikeldking) [inline tests/datasets/test_dataset.py:1290]: datetime format is obviously different in different parts of the world. We should probably make the locale deterministic in the tests.

File: tests/datasets/test_dataset.py (hunk 1 above)
Comment (RogerHYang) [reply tests/datasets/test_dataset.py:1292]: this test is for raising exceptions on invalid entries, so this is fine

**Comment by mikeldking** [comment]
Hey @pbadhe ! Awesome to see you here and thanks for the PR! @RogerHYang will take a look. Thanks agian!

**Review by RogerHYang — APPROVED** [review]
LGTM. Thank you for your contribution!

**Comment by axiomofjoy** [comment]
awesome job @pbadhe 😃

**Pull request merged**